src/ComboBox/ComboBox.svelte

```diff
@@ -299,11 +298,18 @@
                 selectedId = filteredItems[highlightedIndex].id;
               }
             } else {
-              open = false;
-              if (filteredItems[0]) {
-                value = itemToString(filteredItems[0]);
-                selectedItem = filteredItems[0];
-                selectedId = filteredItems[0].id;
+              // searching typed value in text list with lowercase
+              const matchedItem =
+                filteredItems.find(
+                  (e) =>
+                    e.text.toLowerCase() === value?.toLowerCase() && !e.disabled
+                ) ?? filteredItems.find((e) => !e.disabled);
+              if (matchedItem) {
+                // typed value has matched or fallback to first enabled item
+                open = false;
+                selectedItem = matchedItem;
+                value = itemToString(selectedItem.text);
```

Comment:
Argument should be the entire item, not just `text`.

Reply:
Thanks, I fixed!